src/app/service/details/ServiceDetails.jsx

```diff
@@ -25,21 +28,66 @@ function ServiceHeader({ children }) {
   );
 }
 
+const LoadingView = () => (
+  <div className="d-flex flex-column">
+    <NavBar title="Services Details" />
+    <p>
+      <i className="fa fa-spinner fa-spin" aria-hidden="true" /> Loading location data ...{' '}
+    </p>
+  </div>
+);
+
 class ServiceDetails extends Component {
+  state = { service: {} };
+
+  componentWillMount() {
+    const { location } = this.props;
+    if (location) {
+      const service = this.getServiceById(location);
+      this.setState({ service });
+    } else {
+      const { locationId } = this.props.match.params;
+      this.props.getLocation(locationId);
+    }
+  }
+
+  componentWillReceiveProps(nextProps, prevState) {
+    if (nextProps.location !== this.props.location) {
+      const service = this.getServiceById(nextProps.location);
+      this.setState({ service });
+    }
+  }
+
   onGoToDocs = () => {
     const { locationId, serviceId } = this.props.match.params;
     this.props.history.push(`${getServiceUrl(locationId, serviceId)}/documents`);
   };
 
+  getServiceById = (location) => {
```

Comment:
Didn't you say the reducer would just take care of this?

Reply:
I can move this to the reducer 👍